feat(map): Add cycling raster layer from CyclOSM and a visibility toggle

wkyoshida opened the pull request:
<!---
Thank you for your pull request! 🚀
-->

### Contributor checklist

<!-- Please replace the empty checkboxes [] below with checked ones [x] accordingly. -->

- [X] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

<!--
Describe briefly what your pull request proposes to change. Especially if you have more than one commit, it is helpful to give a summary of what your contribution is trying to solve.

Also, please describe shortly how you tested that your change actually works.
-->

Discovered that the URLs available for [CyclOSM](https://www.cyclosm.org) are in the following format:
`https://{switch}.tile-cyclosm.openstreetmap.fr/[cyclosm|cyclosm-lite]/{zoom}/{x}/{y}.png`

- The `cyclosm-lite` version is nice in that it only contains the bike paths in the rendered layers.
- The full `cyclosm` version does highlight bike paths and other cycling infrastructure, but it does render a full "map" - so to speak.

With `cyclosm-lite` though, we can simply overlay it on top of the default layer that we currently use to include the bike paths on top of the existing map.

The property below (L210) configures the layer to not be initially visible, but toggling to the bike profile activates the layer visibility. 
```
  layout: {
      visibility: "none"
  },
```

### Related issue

<!--- activist prefers that pull requests be related to already open issues. -->
<!--- If applicable, please link to the issue by replacing ISSUE_NUMBER with the appropriate number below. -->
<!--- Feel free to delete this section if this does not apply. -->

- #718

andrewtavis commented:
Checking on the idea here, @wkyoshida: you're saying that we can just have the bike paths up when the user has bike routing selected? So then for the layers option it would just be to add in public transit routes, and we wouldn't provide an option to toggle the bike routes? (I think that this might be a good way of going about this, btw!)

wkyoshida commented:
> .. you're saying that we can just have the bike paths up when the user has bike routing selected? So then for the layers option it would just be to add in public transit routes, and we wouldn't provide an option to toggle the bike routes? ..

Yeee - thinking that we could really pretty much just add on the cycling layer and the public transit layer when their respective routing profiles are toggled on.

I'm looking into what our options are for the public transit one though... Transitland also seems like it requires an API key if I understood. Perhaps dealing with a key is what we end up having to do; so might simply be a matter of seeing which provider aligns better and whose free tier is generous :nerd_face: 

---

Quick notes on attribution in regards to 462d1ee: 
- Modified the references linked in the attributions
   - Link instead to https://www.openstreetmap.org/copyright as suggested in the [OSM Attribution Guidelines](https://osmfoundation.org/wiki/Licence/Attribution_Guidelines)
   - For cycling, link also to [CyclOSM](https://www.cyclosm.org) and [OpenStreetMap France](https://openstreetmap.fr)  
- Modified the attributions text to align a little closer to what OSM appears to use:

**BEFORE:**

- Standard

   ![image](https://github.com/user-attachments/assets/79323d1d-8307-433a-8a5d-553cab3f75b7)

- Cycling

   ![image](https://github.com/user-attachments/assets/f30db892-9e48-4ec7-b5b5-aa81bad43807)

**Commit 462d1ee:**

- Standard

   ![image](https://github.com/user-attachments/assets/3cb42fcd-aec0-443e-81a2-74e25a6a50f9)

- Cycling

   ![image](https://github.com/user-attachments/assets/fef90328-8f32-4e62-87fe-1ccda265c256)

https://www.openstreetmap.org:

- Standard

   ![image](https://github.com/user-attachments/assets/3be22e6d-f0a9-4799-a1e2-67501b880207)

- Cycling

   ![image](https://github.com/user-attachments/assets/9578b2c6-894b-49de-ae2e-3b361222b686)

wkyoshida commented on frontend/components/media/MediaMap.vue:
```diff
@@ -167,7 +175,18 @@ onMounted(() => {
+              "cycle-raster-tiles": {
+                type: "raster",
+                tiles: [
+                  "https://a.tile-cyclosm.openstreetmap.fr/cyclosm-lite/{z}/{x}/{y}.png",
+                  "https://b.tile-cyclosm.openstreetmap.fr/cyclosm-lite/{z}/{x}/{y}.png",
+                  "https://c.tile-cyclosm.openstreetmap.fr/cyclosm-lite/{z}/{x}/{y}.png",
```
Adding thoughts here..

It could potentially be good if we're able to load the tile source URLs from env vars - instead of configuring these hard-coded. Same as with `https://tile.openstreetmap.org` for the default layer source URL and the geocoding API URL `https://nominatim.openstreetmap.org`.

To note - as they are, the latter two are probably easy to do as we can simply replace them with an env var each. What about multiple-subdomain URLs though as in here for CyclOSM? We might have to configure a way to  be able to add more or less env vars if the number of subdomains change.

Or - we decide to not concern ourselves too much given that the mentioned services are likely to be and continue fairly stable in the near future? :thinking::shrug:

andrewtavis replied:
Ya I'd say for now let's not concern ourselves too much with this, @wkyoshida :) We can revisit it later when we have a bit more capacity 😊

andrewtavis reviewed (approved):
Really really great, @wkyoshida! Appreciate that the map is continuing to be such a focus and that we're developing something so high quality 😊
